homeassistant/helpers/event.py

```diff
@@ -1587,32 +1586,32 @@ def async_attach(self) -> None:
             f"track time interval {self.seconds}",
             cancel_on_shutdown=self.cancel_on_shutdown,
         )
-        self._cancel_callback = async_call_at(
-            hass,
-            self._track_job,
-            hass.loop.time() + self.seconds,
+        self._schedule_timer()
+
+    def _schedule_timer(self) -> None:
+        """Schedule the timer."""
+        if TYPE_CHECKING:
+            assert self._track_job is not None
+        hass = self.hass
+        loop = hass.loop
+        self._timer_handle = loop.call_at(
+            loop.time() + self.seconds, self._interval_listener, self._track_job
```

Comment:
Its kinda strange to still pass the job here even though we don't use it anymore, but if we take it out it won't be cancelled automatically at shutdown because the we need the `cancel_on_shutdown` logic.